File: src/main/kotlin/com/leetcode/P3137.kt
package com.leetcode

// https://github.com/antop-dev/algorithm/issues/569
class P3137 {
    fun minimumOperationsToMakeKPeriodic(word: String, k: Int): Int {
        var maxCount = 0
        val hash = mutableMapOf<String, Int>() // <부분 문자열, 개수>
        for (i in word.indices step k) {
            val substring = word.substring(i, i + k)
            val count = hash[substring]?.let { it + 1 } ?: 1
            maxCount = maxOf(maxCount, count)
            hash[substring] = count
        }
        val total = word.length / k
        return total - maxCount
    }
}

File: src/python3/com/leetcode/p3137.py
import collections


class Solution:
    def minimumOperationsToMakeKPeriodic(self, word: str, k: int) -> int:
        length = len(word)
        counter = collections.defaultdict(lambda: 0)
        max_count = 0
        for i in range(0, length, k):
            substring = word[i:i + k]
            count = counter[substring] + 1
            max_count = max(max_count, count)
            counter[substring] = count

        total = length // k
        return total - max_count
